ext/recaptcha/CRM/Utils/ReCAPTCHA.php

```diff
@@ -107,6 +107,83 @@ public static function enableCaptchaOnForm(&$form) {
     }
   }
 
+  /**
+   * This checks the form criteria to see if reCAPTCHA should be added and then it adds it to the form if required.
+   *
+   * @param string $formName
+   * @param CRM_Core_Form $form
+   *
+   * @throws \API_Exception
+   * @throws \Civi\API\Exception\UnauthorizedException
+   */
+  public static function checkAndAddCaptchaToForm($formName, &$form) {
+    $addCaptcha = FALSE;
+    $ufGroupIDs = [];
+
```

Comment:
@mattwire I think there are a few common patterns in this function definition, handled under each switch-cases, which can be avoided by: 
1. Removing all `if (!CRM_Core_Session::getLoggedInContactID())` check by: 
```php
  public static function checkAndAddCaptchaToForm($formName, &$form) {
    if (CRM_Core_Session::getLoggedInContactID()) {
       return;
    }
...
```
  2. Getting the $ufGroupIDs assignment in one place : 
  ```php
    public static function checkAndAddCaptchaToForm($formName, &$form) {
        if (CRM_Core_Session::getLoggedInContactID()) {
           return;
        }
       $ufGroupIDs  = method_exists($form, 'getUFGroupIDs') ? $form->getUFGroupIDs() : [];
       ...
  ```

Reply:
@monishdeb I agree with those changes but that would change the existing behaviour which I'm trying to avoid for initial move of the code. As you can see there are inconsistencies - not all forms check if the contact is logged in for example when they probably should.

Reply:
Hmm agree.

Reply:
Actually @monishdeb you were right - I've moved that to a "standard conditions" check.